design/multi-factor-admin-and-step-up.md

```diff
@@ -0,0 +1,582 @@
+-----------------------         |                    |    ---------------
+| OTP Credential 3    |---------|                    |
+-----------------------         ----------------------
+```
+
+This schema serves as a base for all credentials and allows them to be
+used for the appropriate authentication executions. This link allows Keycloak to present
+alternative credentials for an execution during login.
+
+To simplify selection in the GUI, it would also be useful to do some refactoring to ensure
+that the execution code for the `Browser` / `Registration Flow` / `Direct Grant Flow` /
+`Reset Credentials` / `Client Authentication` Bindings can only be selected as executions
+for their binding categories. This already seems to be the case for `Client Authentication`,
+and would simplify the selection of authentication executions within a authentication
+factor.
+
+It may also allow a simplification of the naming when selecting an authentication
+category within an authentication factor. For example, we could just
+have OTP instead of OTP Form (for `Browser` binding) and OTP (for `Direct Grant`)
+
+In addition to its function, an authentication category will also have a set of
+data/metadata that describes it. This can be a general field that will be found
+in all authenticator categories, for example description information (internationalised),
+an icon or something more functional like data for OIDC's amr claims (see section on
+step-up) or whether the authentication category can have one or many authenticators
+assigned to it. This information would not be held in the database, but rather
+directly in Keycloak as static information in the code or as resources.
+
+### Authentication section in the admin console
+
+#### Flows
+
+The schema described in the logic would be implemented
+in the _Authentication > flows_ section. Some work can be done on the UX to make
+the process more intuitive, but this will be proposed at a later state in the
+design process.
+
+##### Default Flows
+
+Keycloak should have the same default flows as is currently the case, except that two
+authentication factors are used, with preset values for the authenticator executions:
+rather than adding and removing authenticator executions, the user can enable or disable
+them.
+This is only UX, because behind Keycloak sets or removes the **disabled** flag within the
+authentication execution's config. The 2nd authentication factor can also be disabled, and
+if either the 2nd factor is disabled, or if all authentication executions are disabled in
+the second factor, then the login will be single factor only.
+
+For example, for the browser flow this would be:
+```
+Auth type                         | Options
+---------------------------------------------------------------------------
+Identity Provider Redirector
+Authenticated on               
+  |- Cookie
+  |- Kerberos                      [ ] enabled  [x] disabled
```

Comment:
Initially, I didn't put them because it wasn't the focus of the explanation. However, to me it may be worthwhile evaluating how useful alternative / required / optional / disabled are in the multi-factor authentication flow, and how they should be used.

Reply:
> it wasn't the focus of the explanation

I see.
This is a breaking change, and I suppose it is unacceptable for existing user.